master/internal/rm/kubernetesrm/log.go

```diff
@@ -6,69 +6,49 @@ import (
 	"time"
 
 	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/sirupsen/logrus"
 
 	"github.com/pkg/errors"
 
 	k8sV1 "k8s.io/api/core/v1"
 	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
 
 	"github.com/determined-ai/determined/master/internal/sproto"
-	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/master/pkg/aproto"
 	"github.com/determined-ai/determined/master/pkg/model"
 )
 
-type (
-	streamLogs struct{}
-)
-
 type podLogStreamer struct {
-	podHandler *actor.Ref
-	logReader  io.ReadCloser
-
-	ctx *actor.Context
+	syslog    *logrus.Entry
+	logReader io.ReadCloser
+	callback  func(log sproto.ContainerLog)
```

Comment:
thoughts about just passing in an `io.Writer` instead of using a callback?

Reply:
if we go that way, `podLogStreamer` doesn't even need to be aware of what a `sproto.ContainerLog` is and we can push that up into pod, maybe further from there.

Reply:
I moved the `sproto.ContainerLog` part to pod, I was curious if our goal was to get rid of those message structures also. I like having the Writer in the log code a little better because it feels more obvious there (where writing to a pod feels like recording a pod configuration), and allows the pod to more logically use multiple writers if needed